crates/sage/src/scoring.rs

```diff
@@ -156,12 +163,45 @@ fn lnfact(n: u16) -> f64 {
     }
 }
 
+fn log_factorial(n: u16, k: u16) -> f64 {
+
+    let k = max(k, 2);
+    let mut result = 0.0;
+
+    for i in (k..=n).rev() {
+        result += (i as f64).ln();
+    }
+
+    result
+}
+
+/// Calculate the OpenMS hyperscore, adds up contributions from matched b and y ions instead of
+/// multiplying them
+fn openms_hyper_score(matched_b: u16, matched_y: u16, summed_b: f32, summed_y: f32) -> f64 {
```

Comment:
Please make this an associated function of the `ScoreType` enum.

For example:
```rs

impl ScoreType {
    pub fn score(self) -> f32 {
        match self {
            Self::SageHyperScore => {
                let i = (summed_b + 1.0) as f64 * (summed_y + 1.0) as f64;
                score = i.ln() + lnfact(matched_b) + lnfact(matched_y)
            }
            Self::OpenMSHyperScore => {
                ...
            }
        }
    }
}
```

Reply:
Replaced it accordingly, hope its correct now.